internal/backend/db/dbgorm/values.go

```diff
@@ -0,0 +1,73 @@
+package dbgorm
+
+import (
+	"context"
+	"errors"
+
+	"gorm.io/gorm"
+
+	"go.autokitteh.dev/autokitteh/internal/backend/db/dbgorm/scheme"
+	"go.autokitteh.dev/autokitteh/internal/kittehs"
+	"go.autokitteh.dev/autokitteh/sdk/sdkerrors"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+func (db *gormdb) UpdateValue(ctx context.Context, envID sdktypes.EnvID, key string, f func(sdktypes.Value) (sdktypes.Value, error)) error {
+	if !envID.IsValid() {
+		return sdkerrors.NewInvalidArgumentError("invalid env_id")
+	}
+
+	return db.transaction(ctx, func(tx *tx) error {
+		v, err := getValue(ctx, &tx.gormdb, envID, key)
+		if err != nil && !errors.Is(err, gorm.ErrRecordNotFound) {
+			return err
+		}
+
+		if v, err = f(v); err != nil {
```

Comment:
why is this a callback based update instead of just getting the value and setting it?

Reply:
atomic operations inside a transaction.